### PR TITLE
[Fix] bug fixing on FLASHDeconv

### PR DESCRIPTION
**Pull request by JeeH-K**
## Checklist
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/openms/include/OpenMS/ANALYSIS/TOPDOWN/PeakGroup.h [hunk 1]
```diff
@@ -103,9 +103,10 @@ namespace OpenMS
            @brief Update isotope cosine sore and qscore. Mono mass is also updated one last time. SNR, per charge SNR, and avg errors are updated here.
            @param avg precalculated averagine
            @param min_cos the peak groups with cosine score less than this will have Qscore 0.
+           @param allowed_isotope_error this set the allowed isotope error in dummy mass generation.
```
Comment (jpfeuffer) [inline src/openms/include/OpenMS/ANALYSIS/TOPDOWN/PeakGroup.h:106]: Can you elaborate a little? What unit is this error? Abosulte or relative?

File: src/openms/source/ANALYSIS/TOPDOWN/FLASHDeconvAlgorithm.cpp [hunk 2]
```diff
@@ -776,7 +701,7 @@ namespace OpenMS
     deconvolved_spectrum_.reserve(mass_bins_.count());
     Size mass_bin_index = mass_bins_.find_first();
     auto peak_bin_numbers = std::vector<Size>(log_mz_peak_size);
-
+    const double max_signal_mz_delta = .16;
```
Comment (jpfeuffer) [inline src/openms/source/ANALYSIS/TOPDOWN/FLASHDeconvAlgorithm.cpp:704]: Can you add a comment on why .16?

File: src/openms/source/ANALYSIS/TOPDOWN/FLASHDeconvAlgorithm.cpp [hunk 3]
```diff
@@ -1022,8 +951,7 @@ namespace OpenMS
         }
       }
 
-
-      if (total_signal_intensity > *std::max_element(total_harmonic_intensity.begin(), total_harmonic_intensity.end()) * 2.0) //
+      if (total_signal_intensity > 1.0 * *std::max_element(total_harmonic_intensity.begin(), total_harmonic_intensity.end())) //
```
Comment (jpfeuffer) [inline src/openms/source/ANALYSIS/TOPDOWN/FLASHDeconvAlgorithm.cpp:954]:
```suggestion
      if (total_signal_intensity > *std::max_element(total_harmonic_intensity.begin(), total_harmonic_intensity.end())) //
```

Comment (jpfeuffer) [reply src/openms/source/ANALYSIS/TOPDOWN/FLASHDeconvAlgorithm.cpp:954]: If this is not the same, please add a comment.

Comment (KyowonJeong) [reply src/openms/source/ANALYSIS/TOPDOWN/FLASHDeconvAlgorithm.cpp:954]: They are. I tried with different values beforehand and 1.0 is the most reasonable value. Before the calculation was not correct and I used 2.0 and now after fixing the bug 1.0 resulted in the best deconvolution.

File: src/openms/include/OpenMS/ANALYSIS/TOPDOWN/PeakGroup.h (hunk 1 above)
Comment (KyowonJeong) [reply src/openms/include/OpenMS/ANALYSIS/TOPDOWN/PeakGroup.h:106]: Oh. That is the number.. Like one isotope or two. I will add comment

**Review by jpfeuffer — COMMENTED** [review]
Rest looks good. Only little comments.

File: src/openms/include/OpenMS/ANALYSIS/TOPDOWN/PeakGroup.h (hunk 1 above)
Comment (jpfeuffer) [reply src/openms/include/OpenMS/ANALYSIS/TOPDOWN/PeakGroup.h:106]: Ah yes, that would be good. Not sure how to name it, "isotope number" might be ok, yes.

**Pull request merged**